src/middleware/express-error.js

```diff
@@ -0,0 +1,70 @@
+const uuid = require('uuid');
+
+/**
+ * Create parameter validation middle-ware instance, given a mapping from
+ * parameter to regular expression or function that returns a message as string
+ * if the parameter is invalid.
+ *
+ * Parameters not listed in `req.params` will be ignored. But parameters
+ * present must match the pattern given in `options` or the request will be
+ * rejected with a 400 error message.
+ */
+const expressError = ({errorCodes, entry, monitor}) => {
+  const {name: method, cleanPayload} = entry;
+  return (err, req, res, next) => {
```

Comment:
I don't see a special-case for the ErrorReply type here (`err instanceof ErrorReply`).  Instead, it depends on the fact that such objects have `code`, `details`, and `message` -- but lots of errors have `code` and `message`, and it's not unreasonable to think one might have `details`.  This would result in showing that data to the caller, possibly revealing secret information in `details`.

A better separation would be something like this:

```js
if (!err instanceof ErrorReply) {
  // convert an arbitrary error into an ErrorReply, obscuring any details of the error
  // as they might reveal useful information to an attacker
  // .. call monitor.reportError like the old reportInternalError did
  // .. create an incidentId and set `err` to a new ErrorReply similar to how reportInternalError called reportError
}
// .. continue on as below, now knowing that `err` is an ErrorReply object
```